### PR TITLE
adding support for V2 Compat Mode

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: src/WebJobs.Script/Config/CompatModeOptions.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    internal class CompatModeOptions
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/CompatModeOptions.cs]: nit: rename -> `CompatibilityModeOptions`

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs [hunk 2]
```diff
@@ -86,6 +86,13 @@ public static bool IsCoreTools(this IEnvironment environment)
             return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(CoreToolsEnvironment));
         }
 
+        public static bool IsFunctionsV2CompatibilityMode(this IEnvironment environment)
```
Comment (fabiocav) [inline src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: nit: Make this consistent with the property name (drop `Functions` or add `Functions` to the property)

File: src/WebJobs.Script/Environment/EnvironmentSettingNames.cs [hunk 3]
```diff
@@ -32,6 +32,7 @@ public static class EnvironmentSettingNames
         public const string ConsoleLoggingDisabled = "CONSOLE_LOGGING_DISABLED";
         public const string SkipSslValidation = "SCM_SKIP_SSL_VALIDATION";
         public const string EnvironmentNameKey = "AZURE_FUNCTIONS_ENVIRONMENT";
+        public const string FunctionsV2CompatibilityModeKey = "FUNCTIONS_V2_COMPATIBILITY_MODE";
```
Comment (fabiocav) [inline src/WebJobs.Script/Environment/EnvironmentSettingNames.cs]: Same comment about name consistency

File: src/WebJobs.Script/ScriptHostBuilderExtensions.cs [hunk 4]
```diff
@@ -190,6 +190,8 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
         public static void AddCommonServices(IServiceCollection services)
         {
+            services.ConfigureOptions<CompatModeOptionsSetup>();
```
Comment (fabiocav) [inline src/WebJobs.Script/ScriptHostBuilderExtensions.cs]: `Compat` -> `Compatibility`

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs (hunk 2 above)
Comment (brettsam) [reply src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: I've removed `Functions`. It was getting long when you went to `FunctionsCompatibilityModeChangeTokenSource`...

**Pull request merged**